docs/parse_argparse.py

```diff
@@ -0,0 +1,118 @@
+import glob
+import subprocess
+import os
+import textwrap 
+import shutil
+
+path_scripts_2 = '../docs/source/scripts_2'
+if os.path.exists(path_scripts_2):
+    shutil.rmtree(path_scripts_2) 
+
+for dir_name in glob.glob("../autometa/**/*.py", recursive=True): 
+    if "__" not in dir_name: #to not include __init__ files
+        print(dir_name)
+        with open (dir_name) as fh_read:
+            path_temp_write = '../docs/temp_write.py'
+            fh_write = open(path_temp_write, 'w+')            
+            for line in fh_read:
+                if line.strip() == 'import argparse':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+            # Reads text until the end of the block:
+            for line in fh_read:  # This keeps reading the file
+                if line.strip() == 'args = parser.parse_args()':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+                line = line.strip()
+                fh_write.write(line + '\n')
+            fh_write.close()
+        
+        basename = os.path.basename(dir_name) #Extract the filename, eg: kmers.py
```

Comment:
Need a little help on this one. Since I am copying the whole `argparse` block, how should I insert the `usage` information in between while writing it?

Reply:
While you are iterating through each file from which you are capturing usage information, you will be within the context of interacting with one particular file at a time.

e.g. 
```
fpaths = glob.glob(...)
for fpath in fpaths:
    # We are now within the context of interacting with only one file
    filename = os.path.basename(fpath)
    # Now we have the filename from which we are capturing the usage information
    ...
    # get the argparse block and during grabbing the argparse block you will
    argparse_block = get_argparse_block(...)
    # at some point you will encounter the parser line:
    parser_line = "parser = argparse.ArgumentParser(description=...)"
    usage = f"usage={filename}, description"
    newline = parser_line.replace('description', usage)
```

This should help point you on the right track.